MinimalLifts.m

```diff
@@ -0,0 +1,231 @@
+MaximalOrder:=function(G)
+  x:=[z:z in Generators(G)];
+a:=x[1];
+n:=Order(G);
+i:=1;
+while Order(a) ne n do
+	   i:=i+1;
+a:=a*x[i];
+end while;
+return a;
+end function;
+
+
+CentralProductPC:=function(n,H,Z)
+  T:=H meet Z;
+k1:=Order(T);
+if Order(Z) eq 1 then
+return H;
+end if;
+p:=Order(H);
+k2:=n*k1;
+k:=k2/p;
+k:=Numerator(k);
+C:=PCGroup(CyclicGroup(k));
+B, inc:=DirectProduct(H, C);
+x:=MaximalOrder(Z);
+y:=MaximalOrder(C);
+z:=Order(Z);
+a:=z/k1;
+a:=Numerator(a);
+b:=k/k1;
+b:=Numerator(b);
+N:=sub<B|inc[1](x^a)*inc[2](y^b)>;
+return B / N;
+end function;
+
+
+CentralProductPerm:=function(n,H,Z)
```

Comment:
Given the length of this function, it would be better to use `Type(H)` to merge it with the last function.  You can check whether H is a PC group or a permutation group, and then make the appropriate kind of cyclic group.